globals.go

```diff
@@ -1084,3 +1084,55 @@ const (
 	LogViewerHTML    LogViewer = "html"
 	LogViewerLobster LogViewer = "lobster"
 )
+
+// ContainerOS denotes the operating system of a running container. Only linux and windows are supported.
+type ContainerOS string
+
+const (
+	LinuxOS   = "linux"
+	WindowsOS = "windows"
+)
+
+func (c ContainerOS) Validate() error {
+	switch c {
+	case LinuxOS, WindowsOS:
+		return nil
+	default:
+		return errors.Errorf("unrecognized container OS '%s'", c)
+	}
+}
+
+// CPUArchitecture represents the architecture property of a container required to provide to Amazon ECS
+type CPUArchitecture string
+
+const (
+	ArchARM64 = "arm64"
+	ArchAMD64 = "x86_64"
+)
+
+func (c CPUArchitecture) Validate() error {
+	switch c {
+	case ArchARM64, ArchAMD64:
+		return nil
+	default:
+		return errors.Errorf("unrecognized CPU architecture '%s'", c)
+	}
+}
+
+// WindowsVersion specifies the Windows version of a container if running on Windows device (required if os is windows)
```

Comment:
Period at the end of comment/documentation sentences. I don't think the comment in parentheses is necessary since it's already stated that it's running on a Windows machine. I would phrase the comment as something like: "WindowsVersion specifies the compatibility version of Windows required for the container to run."